### PR TITLE
Add GraalVM Native Image docs

### PR DESCRIPTION
**Pull request by beckermarc**
Merge only, after internal changes have been done.

### CODE REVIEW

File: java/development/index.md [hunk 1]
```diff
@@ -371,7 +433,11 @@ If you want to have the comfort of an automated CDS build like with the `watch`
 If the Spring Boot Devtools configuration of your CAP Java application defines a [trigger file](https://docs.spring.io/spring-boot/docs/current/reference/html/using.html#using.devtools.restart.triggerfile), the `auto-build` can detect this and touch the trigger file in case of any file change. The same applies to the `watch` goal.
 :::
 
-<div id="afterautobuild" />
+#### Local Development for Multitenant Applications
+
+With the streamlined MTX, you can run your multitenant application locally along with the MTX sidecar and use SQLite as the database. See [the _Deploy as SaaS_ guide](../../guides/deployment/as-saas#local-mtx) for more information.
```
Comment (renejeglinsky) [inline java/development/index.md:438]: We could even add the toggle.
```suggestion
With the streamlined MTX, you can run your multitenant application locally along with the MTX sidecar and use SQLite as the database. See [the _Deploy as SaaS_ guide](../../guides/deployment/as-saas?impl-variant=java#local-mtx) for more information.
```

File: java/development/index.md [hunk 2]
```diff
@@ -141,15 +141,15 @@ It might be more convenient to make use of CDS starter bundle `cds-starter-sprin
...
 ### Spring Features
 
-Beside the common Spring features such as dependency injection and a sophisticated test framework, the following features are available in Spring CAP applications in addition:
+Beside the common Spring features such as dependency injection and a sophisticated test framework, the following features are available in Spring CAP applications:
 
 * CDS event handlers within custom Spring beans are automatically registered at startup.
 * Full integration into Spring transaction management (`@Transactional` is supported).
-* A various number of CAP Java SDK interfaces are exposed as [Spring beans](#exposed-beans) and are available in Spring application context such as technical services, the `CdsModel` or the `UserInfo` in current request scope.
-* *Automatic* configuration of XSUAA, IAS and [mock user authentication](../security#mock-users) by means of Spring security configuration.
+* A number of CAP Java SDK interfaces are exposed as [Spring beans](#exposed-beans) and are available in Spring application contexts such as technical services, the `CdsModel`, or the `UserInfo` in current request scope.
```
Comment (beckermarc) [inline java/development/index.md:148]: There is only one spring application context in an application.

```suggestion
* A number of CAP Java SDK interfaces are exposed as [Spring beans](#exposed-beans) and are available in the Spring application context such as technical services, the `CdsModel`, or the `UserInfo` in current request scope.
```

File: java/development/index.md [hunk 3]
```diff
@@ -183,14 +183,16 @@ If you want to compile your application as a native executable the following bou
...
+2. Dynamic elements of your code, such as usage of reflection, JDK proxies, or resources need to be registered with the GraalVM Native Image build. You can learn more about this in the [GraalVM Native Image documentation](https://www.graalvm.org/latest/reference-manual/native-image/metadata/).
 
     ::: tip
-    Many runtime hints for reflection, JDK proxy usage and resources are contributed automatically to the Native Image build.
-    This includes required reflection for event handler classes defined in application code and required JDK proxy usage for interfaces generated from the application's CDS model by the CDS Maven Plugin.
+    Many runtime hints for reflection, JDK proxy usage, and resources are contributed automatically to the Native Image build.
+    This includes 
+    - Required reflection for event handler classes defined in application code
+    - The usage of a required JDK proxy for interfaces generated from the application's CDS model by the CDS Maven Plugin.
```
Comment (beckermarc) [inline java/development/index.md:192]:
```suggestion
    - The usage of JDK proxies for interfaces generated from the application's CDS model by the CDS Maven Plugin.
```

**Review by smahati — APPROVED** [review]
I think it's good, if it's ok with @beckermarc you can merge it.

**Pull request merged**